# HG changeset patch
# User Pantxo Diribarne <pantxo.diribarne@gmail.com>
# Date 1427097819 -3600
#      Mon Mar 23 09:03:39 2015 +0100
# Node ID 5e6e0563388b26d9180bfcff2c517e4876325e02
# Parent  f902855f870ac4170eeee625d1e91b5a1d5811f5
Handle "auto" value of figure "toolbar" property (bug #44573)

* Figure.cc (Figure::update): when handling a change in "menubar" or "toolbar" property, take toolbar "auto" value into account.

diff --git a/libgui/graphics/Figure.cc b/libgui/graphics/Figure.cc
--- a/libgui/graphics/Figure.cc
+++ b/libgui/graphics/Figure.cc
@@ -435,21 +435,24 @@ Figure::update (int pId)
       break;
 
     case figure::properties::ID_TOOLBAR:
       if (fp.toolbar_is ("none"))
         showFigureToolBar (false);
       else if (fp.toolbar_is ("figure"))
         showFigureToolBar (true);
       else // "auto"
-        showFigureToolBar (! hasUiControlChildren (fp));
+        showFigureToolBar (! hasUiControlChildren (fp) &&
+                           fp.menubar_is ("figure"));
       break;
 
     case figure::properties::ID_MENUBAR:
       showMenuBar (fp.menubar_is ("figure"));
+      if (fp.toolbar_is ("auto"))
+        showFigureToolBar (fp.menubar_is ("figure"));
       break;
 
     case figure::properties::ID_KEYPRESSFCN:
       if (fp.get_keypressfcn ().is_empty ())
         m_container->canvas (m_handle)->clearEventMask (Canvas::KeyPress);
       else
         m_container->canvas (m_handle)->addEventMask (Canvas::KeyPress);
       break;
